Upgrading from version 2.0.1 to 2.1.0 or 2.2.0 breaks vite build with TS (TS7006) errors

Hi Team

I just upgraded to version 2.2.0 and the change creates those this really weird issue, I am not changing anything on my tsconfig and eslinyrc and suddenly running vite build throws more than 600 implicit any type

Have anyone seen these before?

I had a similar error 

```
node_modules/@reduxjs/toolkit/dist/createDraftSafeSelector.d.ts:10:47 - error TS2707: Generic type 'CreateSelectorFunction<MemoizeFunction, ArgsMemoizeFunction>' requires between 0 and 2 type arguments.

10 export declare const createDraftSafeSelector: import("reselect").CreateSelectorFunction<typeof weakMapMemoize, typeof weakMapMemoize, any>;
```

and was able to fix it by updating the `reselect` dependency (from v5.0.1 to 5.1.0 in my case).

Similar to @AlbericoD I just removed and re-added rtk...

```bash
yarn remove @reduxjs/toolkit
yarn add @reduxjs/toolkit
```

--- a/packages/toolkit/package.json
+++ b/packages/toolkit/package.json
@@ -119,7 +119,7 @@
     "immer": "^10.0.3",
     "redux": "^5.0.1",
     "redux-thunk": "^3.1.0",
-    "reselect": "^5.0.1"
+    "reselect": "^5.1.0"
   },
   "peerDependencies": {
     "react": "^16.9.0 || ^17.0.0 || ^18",

